hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/EventloopBuilderTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.tpc;
+
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+
+public abstract class EventloopBuilderTest {
+
+    public abstract EventloopBuilder create();
+
+    @Test(expected = NullPointerException.class)
+    public void test_setThreadFactory_whenNull() {
+        EventloopBuilder factory = create();
```

Comment:
Minor: We can rename factory to builder. Same for other tests

Reply:
Done.